testing/src/test/java/org/motechproject/nms/testing/it/api/MobileAcademyControllerBundleIT.java

```diff
@@ -1003,5 +1128,210 @@ public void verifyFT532() throws IOException, InterruptedException {
         assertTrue("{\"bookmark\":\"Chapter01_Lesson01\",\"scoresByChapter\":{\"1\":2}}"
                 .equals(EntityUtils.toString(response.getEntity())));
     }
+
+    /**
+     * To check delivery notification API is successful when optional parameter
+     * call data is missing.
+     */
+    // TODO https://applab.atlassian.net/browse/NMS-250
+    @Ignore
```

Comment:
There is a fix coming in for NMS-250 but this test needs a different design. Sms status updates are handled asynchronously while this test tries to test it synchronously, causing the completion record to be read before the event handler code updates the status.

You should either sleep here or to be even more reliable, test the different layers (api vs update) differently.